hbase_sql/src/main/java/com/splicemachine/olap/OlapServerMaster.java

```diff
@@ -205,10 +176,12 @@ public void run() {
         publishServer(rzk, serverName, hostname, port);
 
         SpliceSpark.getContextUnsafe(); // kickstart Spark
-        
+
         while(!end.get()) {
             Thread.sleep(10000);
             ugi.checkTGTAndReloginFromKeytab();
+            if (yarnUgi != null)
+                yarnUgi.checkTGTAndReloginFromKeytab();
         }
```

Comment:
Why need to login twice using the same principal/keytab and relogin from both UGI?

Reply:
Do we know why it didn't work?